ts/Dashboards/Plugins/HighchartsComponent.ts

```diff
@@ -719,6 +717,36 @@ class HighchartsComponent extends Component {
         return this.chart || this.createChart();
     }
 
+    /**
+     * Creates default mapping when columnAssignment is not declared.
+     * @param  { Array<string>} columnNames all columns returned from dataTable.
+     *
+     * @returns
+     * The record of mapping
+     *
+     * @private
+     *
+     */
+    private getDefaultColumnAssignment(
+        columnNames: Array<string> = []
+    ): Record<string, string | null> {
+        const defaultColumnAssignment:Record<string, string> = {};
+
+        for (let i = 0, iEnd = columnNames.length; i < iEnd; ++i) {
+            defaultColumnAssignment[columnNames[i]] = 'y';
+            if (i === 0) {
+                const firstColumnValues =
+                    this.presentationTable?.getColumn(columnNames[i]);
```

Comment:
If you just test the first cell of the column, then it is more efficient to use `DataTable.getCell`.

Reply:
Otherwise you can also use the asReference argument of `DataTable.getColumn`, so that you directly access the actual column instead of a slice copy.